pyk/src/pyk/kast/manip.py

```diff
@@ -854,3 +861,30 @@ def _no_cell_rewrite_to_dots(_term: KInner) -> KInner:
     subst = Subst({cell_name: _no_cell_rewrite_to_dots(cell_contents) for cell_name, cell_contents in _subst.items()})
 
     return subst(config)
+
+
+def defunctionalize(defn: KDefinition, kinner: KInner) -> tuple[KInner, list[KInner]]:
```

Comment:
I think this could even be unit tested, the only tedious part is setting up a definition. Here's an example from another unit test: https://github.com/runtimeverification/k/blob/809d59eb819dcb14ac5fb0eb9abe1c05eae38fde/pyk/src/tests/unit/kast/test_formatter.py#L34-L47

Reply:
Done, I added them to `unit/kast/test_manip.py`, and tested a bunch of scenarios. I kept the integration test I added as well.